dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -1001,6 +1003,18 @@ private void initializeLocalWorker() throws IOException {
     mLocalWorkerInitialized = true;
   }
 
+  /**
+   * Creates an underlying file system which handles UFS fallback.
+   *
+   * @param ufsOptions options to access the UFS
+   * @return a UFS-based FileSystem implementation
+   */
+  public FileSystem createUfsBaseFileSystem(Optional<UfsFileSystemOptions> ufsOptions) {
```

Comment:
I think you can just accept the argument without the `Optional` part, and let the callers to make sure they do have a `UfsFileSystemOptions`.

Reply:
The callers typically have more context in handling an empty `Optional<UfsFileSystemOptions>` than `FileSystemContext`, e.g. using a default, or throwing an error to alert the user.